jib-core/src/main/java/com/google/cloud/tools/jib/builder/steps/StepsRunner.java

```diff
@@ -430,89 +434,138 @@ private void checkImageInTargetRegistry() {
-                  : manifestPushResults.get(0).get();
+                                      .get(baseImage)));
+
+                          List<Future<BuildResult>> manifestPushResults =
+                              scheduleCallables(
+                                  PushImageStep.makeList(
+                                      buildContext,
+                                      childProgressDispatcherFactory,
+                                      results.targetRegistryClient.get(),
+                                      Verify.verifyNotNull(
+                                              results
+                                                  .builtImagesAndContainerConfigurationPushResults
+                                                  .get()
+                                                  .get(builtImage))
+                                          .get(),
+                                      builtImage.get(),
+                                      results.manifestCheckResult.get().isPresent()));
+
+                          realizeFutures(manifestPushResults);
+                          return manifestPushResults.isEmpty()
+                              ? new BuildResult(
+                                  results.manifestCheckResult.get().get().getDigest(),
+                                  Verify.verifyNotNull(
+                                          results
+                                              .builtImagesAndContainerConfigurationPushResults
+                                              .get()
+                                              .get(builtImage))
+                                      .get()
+                                      .getDigest())
+                              // Manifest pushers return the same BuildResult.
+                              : manifestPushResults.get(0).get();
+                        });
+
+                buildResults.add(buildResult);
+              }
+              return buildResults;
             });
   }
 
   private void loadDocker(DockerClient dockerClient) {
     ProgressEventDispatcher.Factory childProgressDispatcherFactory =
         Verify.verifyNotNull(rootProgressDispatcher).newChildProducer();
 
-    results.buildResult =
+    results.buildResults =
         executorService.submit(
-            () ->
-                new LoadDockerStep(
-                        buildContext,
-                        childProgressDispatcherFactory,
-                        dockerClient,
-                        results.builtImages.get().get(0).get())
-                    .call());
+            () -> {
+              Map.Entry<Future<Image>, Image> entry =
+                  results.builtImagesAndBaseImages.get().entrySet().iterator().next();
+              Future<Image> builtImage = entry.getKey();
+
+              List<Future<BuildResult>> buildResults = new ArrayList<>();
+              Future<BuildResult> buildResult =
+                  executorService.submit(
```

Comment:
Can be simplified by `return Collections.singletonList(executorService.submit(...))`.